content/strategy-goals/strategy/use-cases/code-health.md

```diff
@@ -56,6 +56,15 @@ The following are real, anonymized quotes from customers today on how we're alre
 - Tie to higher retainment of engineers when they work in healthy code environments
 - Increased productivity, quicker builds
 
+## Competitive positioning
+
+As we position on the code health and [fixing security vulnerabilities](fixing-security-vulnerabilities.md) use cases, static analysis products such as Sonarqube and semgrep come up in customer conversations. The expressivity of structural search coupled with Batch Changes, for example, is a powerful approach to find and fix a code smell or vulnerability, which is also something static analysis is good at. So **what is the role of search vs static analysis** for those use cases? Each has their own merit and areas of use:
+
+- **static analysis** is typically harder to write, but is accurate and unambiguous. Static analysis products tend to orient towards **exhaustivity out of the box**: make sure known best practices are in place or make sure to scan for all common security vulnerabilities. They cannot easily be used to iterate on a new or ad hoc question about the codebase, and typically require previous experience with static analysis.
+- **universal code search** is easier to use and easy to iterate with to answer any question aboute the codebase. For example, when a new code health issue arises, it's easy to quickly explore what code it impacts and write a batch change to fix it. It may not be as accurate or unambiguous: describing something unambiguously is not always possible or easy. But search can be used by any developer to very quickly solve relatively complicated problems, without requiring the help of someone with experience with static analysis.
+
+As we increase our focus on those use cases we are not orienting towards exhaustivity: we will not be a product that out of the box scans for all CVEs or automatically applies basic linting rules. However, Sourcegraph will provide templates to solve for common use cases and get users started and more importantly, the expressibity and flexibility of search allows any developer to iterate over any question about the codebase.
```

Comment:
```suggestion
As we increase our focus on those use cases we are not orienting towards exhaustivity: we will not be a product that out of the box scans for all CVEs or automatically applies basic linting rules. However, Sourcegraph will provide templates to solve for common use cases and lower the barrier to learning our features. The purpose of our templates is primarily to provide a starting point that may require small modifications to achieve a valuable impact, rather than something that is usable out of the box without any modifications. Importantly, the expressibility and flexibility of search are what allow any developer to easily modify our templates. 
```

Reply:
I think having out of the box templates that just work is _also_ a great addition, I just don't think we should (a) require that they work out of the box (b) claim that we provide all the templates you need.

Reply:
(I agree, hence clarified "primary purpose" rather than "sole purpose" )